File: TwentyFourDuel.swiftpm/Sources/GameView.swift
import SwiftUI

struct GameView: View {
    @StateObject private var game = GameState()

    private let ticker = Timer.publish(every: 1, on: .main, in: .common).autoconnect()

    var body: some View {
        ZStack {
            Color.bgMain

            switch game.phase {
            case .welcome:
                WelcomeView(game: game)
                    .transition(.opacity)
            case .history:
                HistoryView {
                    withAnimation(.easeInOut(duration: 0.2)) { game.phase = .welcome }
                }
                .transition(.opacity)
            case .playing, .roundOver:
                gamePlayView
                    .transition(.opacity)
            case .gameOver:
                ResultView(game: game)
                    .transition(.opacity)
            }
        }
        .ignoresSafeArea()
        .onReceive(ticker) { _ in game.tick() }
    }

    private let centerHeight: CGFloat = 60

    private var gamePlayView: some View {
        GeometryReader { geo in
            let playerH = (geo.size.height - centerHeight) / 2
            VStack(spacing: 0) {
                PlayerView(game: game, playerIndex: 1)
                    .rotationEffect(.degrees(180))
                    .frame(width: geo.size.width, height: playerH)

                centerStrip
                    .frame(width: geo.size.width, height: centerHeight)

                PlayerView(game: game, playerIndex: 0)
                    .frame(width: geo.size.width, height: playerH)
            }
        }
        .ignoresSafeArea()
    }

    // MARK: - Center Strip

    private var centerStrip: some View {
        HStack(spacing: 12) {
            centerCards
        }
        .frame(maxWidth: .infinity)
        .background(
            LinearGradient(
                colors: [Color.p1Accent.opacity(0.03), Color.p2Accent.opacity(0.03)],
                startPoint: .leading,
                endPoint: .trailing
            )
        )
        .overlay(alignment: .top) {
            Rectangle().fill(Color.white.opacity(0.04)).frame(height: 1)
        }
        .overlay(alignment: .bottom) {
            Rectangle().fill(Color.white.opacity(0.04)).frame(height: 1)
        }
    }

    private var centerCards: some View {
        HStack(spacing: 7) {
            ForEach(0..<game.numbers.count, id: \.self) { i in
                MiniCardView(
                    number: game.numbers[safe: i] ?? 0,
                    suit: game.suits[safe: i] ?? .spade
                )
            }
        }
    }
}

// MARK: - Mini Card (center decorative)

struct MiniCardView: View {
    let number: Int
    let suit: CardSuit

    private var textColor: Color { suit.isRed ? Color(red: 0.8, green: 0, blue: 0) : Color(white: 0.13) }

    var body: some View {
        ZStack {
            RoundedRectangle(cornerRadius: 5)
                .fill(
                    LinearGradient(
                        colors: [.cardGradientLight, .cardGradientDark],
                        startPoint: .topLeading,
                        endPoint: .bottomTrailing
                    )
                )

            Text(suit.symbol)
                .font(.system(size: 13, weight: .semibold))
                .foregroundColor(textColor)

            VStack {
                HStack {
                    Text("\(number)")
                        .font(.system(size: 10, weight: .bold))
                        .foregroundColor(textColor)
                        .padding(.leading, 4).padding(.top, 2)
                    Spacer()
                }
                Spacer()
                HStack {
                    Spacer()
                    Text("\(number)")
                        .font(.system(size: 10, weight: .bold))
                        .foregroundColor(textColor)
                        .rotationEffect(.degrees(180))
                        .padding(.trailing, 4).padding(.bottom, 2)
                }
            }
        }
        .frame(width: 40, height: 54)
        .shadow(color: .black.opacity(0.3), radius: 3, y: 2)
    }
}

File: TwentyFourDuel-Android/app/src/main/java/com/twentyfourduel/ui/GameView.kt
package com.twentyfourduel.ui

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Text
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.draw.rotate
import androidx.compose.ui.draw.shadow
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.twentyfourduel.model.CardSuit
import com.twentyfourduel.model.GamePhase
import com.twentyfourduel.model.GameState
import kotlinx.coroutines.delay

@Composable
fun GameScreen(game: GameState) {
    LaunchedEffect(game.phase) {
        while (true) {
            delay(1000)
            game.tick()
        }
    }

    // Auto-advance after round over
    val phase = game.phase
    LaunchedEffect(phase) {
        if (phase == GamePhase.ROUND_OVER) {
            delay(2500)
            game.advanceAfterRound()
        }
    }

    Box(
        modifier = Modifier.fillMaxSize().background(AppColors.bgMain)
    ) {
        when (game.phase) {
            GamePhase.WELCOME -> WelcomeView(game)
            GamePhase.HISTORY -> HistoryScreen(game)
            GamePhase.PLAYING, GamePhase.ROUND_OVER -> GamePlayView(game)
            GamePhase.GAME_OVER -> ResultView(game)
        }
    }
}

@Composable
private fun GamePlayView(game: GameState) {
    val centerHeight = 60.dp
    Column(modifier = Modifier.fillMaxSize()) {
        Box(
            modifier = Modifier
                .fillMaxWidth()
                .weight(1f)
                .rotate(180f)
        ) {
            PlayerView(game, 1, Modifier.fillMaxSize())
        }

        CenterStrip(game, Modifier.fillMaxWidth().height(centerHeight))

        Box(modifier = Modifier.fillMaxWidth().weight(1f)) {
            PlayerView(game, 0, Modifier.fillMaxSize())
        }
    }
}

@Composable
private fun CenterStrip(game: GameState, modifier: Modifier) {
    Row(
        modifier = modifier
            .background(
                Brush.horizontalGradient(
                    listOf(AppColors.p1Accent.copy(alpha = 0.03f), AppColors.p2Accent.copy(alpha = 0.03f))
                )
            ),
        horizontalArrangement = Arrangement.Center,
        verticalAlignment = Alignment.CenterVertically
    ) {
        for (i in game.numbers.indices) {
            if (i > 0) Spacer(Modifier.width(7.dp))
            MiniCard(
                game.numbers.getOrElse(i) { 0 },
                game.suits.getOrElse(i) { CardSuit.SPADE }
            )
        }
    }
}

@Composable
private fun MiniCard(number: Int, suit: CardSuit) {
    val textColor = if (suit.isRed) Color(204, 0, 0) else Color(33, 33, 33)
    Box(
        modifier = Modifier
            .width(40.dp).height(54.dp)
            .shadow(3.dp, RoundedCornerShape(5.dp))
            .clip(RoundedCornerShape(5.dp))
            .background(Brush.linearGradient(listOf(AppColors.cardGradLight, AppColors.cardGradDark))),
        contentAlignment = Alignment.Center
    ) {
        Text(suit.symbol, fontSize = 13.sp, fontWeight = FontWeight.SemiBold, color = textColor)
        Text(
            "$number",
            fontSize = 10.sp,
            fontWeight = FontWeight.Bold,
            color = textColor,
            fontFamily = FontFamily.Monospace,
            modifier = Modifier.align(Alignment.TopStart).padding(start = 4.dp, top = 2.dp)
        )
        Text(
            "$number",
            fontSize = 10.sp,
            fontWeight = FontWeight.Bold,
            color = textColor,
            fontFamily = FontFamily.Monospace,
            modifier = Modifier.align(Alignment.BottomEnd).padding(end = 4.dp, bottom = 2.dp).rotate(180f)
        )
    }
}
